### PR TITLE
#3930 - Fix for main branch removing api controller attributes which …

### LINKED ISSUE #3930: In 8.1.1 (prob in 9 too) - when ".UseInMemoryApplicationContextManager = true" you get "Unable to resolve service for type 'Csla.State.ISessionManager'"
First noticed by @russblair mentioned on [discord channel](https://discord.com/channels/795719451148156938/795719451148156941)
Confirmed by @swegele

**Repro:**
Blazor Server-Interactive Only project (no wasm or any refs to new .NET 8 state management stuff)
Config in blazor server program.cs 

```csharp
builder.Services.AddCsla(o => o
  .AddAspNetCore()
  .AddServerSideBlazor(bsco => bsco
    .UseInMemoryApplicationContextManager = true));

...

        app.UseEndpoints(delegate (IEndpointRouteBuilder endpoint)
        {
            endpoint.MapControllers();
        });
```

Csla config correctly skips making any reference or registration to any state management stuff.  But none-the-less somehow .NET DI is looking for `ISessionManager`.  We think that ASPNet is scanning the project (and it's dependencies) for controllers and it finds the `[ApiController]` attribute here:

Csla.AspNetCore.Blazor.State.StateController
```csharp
  [ApiController]
  [Route("[controller]")]
  public class StateController(ApplicationContext applicationContext, ISessionManager sessionManager) : ControllerBase
```

**WORKAROUND**
I did a test workaround suggested by @TheCakeMonster making a no-op implementation for the `ISessionManager` interface and registered it.  With this the DI error goes away and project runs as expected.

```csharp
using Csla.State;

    public class NoOpSessionManager : ISessionManager
    {
        public Session GetCachedSession()
        { throw new NotImplementedException(); }
        public Session GetSession()
        { throw new NotImplementedException(); }
        public void PurgeSessions(TimeSpan expiration)
        { throw new NotImplementedException(); }
        public Task<Session> RetrieveSession()
        { throw new NotImplementedException(); }
        public Task SendSession()
        { throw new NotImplementedException(); }
        public void UpdateSession(Session newSession)
        { throw new NotImplementedException(); }
    }
...

builder.Services.AddSingleton<Csla.State.ISessionManager, NoOpSessionManager>();

```

### PR DESCRIPTION
**Pull request by swegele**
…stops ASPNet from including StateController (and it's CTOR services) in service registration; 
also implementing a no-op session manager when in memory app context manager is used.
fixes #3930

### CODE REVIEW

File: Source/Csla.Blazor/State/NoOpSessionManager.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+{
+  /// <summary>
+  /// If <see cref="Configuration.BlazorServerConfigurationOptions.UseInMemoryApplicationContextManager"/> is set to <see langword="true"/> then this class
+  /// will be registered with DI as the default implementation of <see cref="ISessionManager"/>, because in that scenario there should be NO references to any
+  /// of the state or session management mechanisms which support the flow of <see cref="ApplicationContext"/> between Server-Interactive and WASM-Interactive code.
+  /// </summary>
+  public class NoOpSessionManager : ISessionManager
+  {
+    /// <summary>
+    /// Not implemented
```
Comment (StefanOssendorf) [inline Source/Csla.Blazor/State/NoOpSessionManager.cs:21]: Write not supported :)

File: Source/Csla.Blazor/State/NoOpSessionManager.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    { throw new NotSupportedException(); }
+    /// <summary>
+    /// Not implemented
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="NotSupportedException"></exception>
+    public Task<Session> RetrieveSession()
+    { throw new NotSupportedException(); }
+
+    public Task<Session> RetrieveSession(TimeSpan timeout)
```
Comment (StefanOssendorf) [inline Source/Csla.Blazor/State/NoOpSessionManager.cs:49]: xml comment missing. Get rid of the warnings :D

File: Source/Csla.Blazor/State/NoOpSessionManager.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    /// <summary>
+    /// Not implemented
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="NotSupportedException"></exception>
+    public Task SendSession()
+    { throw new NotSupportedException(); }
+
+    public Task SendSession(TimeSpan timeout)
```
Comment (StefanOssendorf) [inline Source/Csla.Blazor/State/NoOpSessionManager.cs:60]: Warnings

**Review by StefanOssendorf — CHANGES_REQUESTED** [review]

**Comment by rockfordlhotka** [comment]
@swegele isn't this tied to an issue? That should be mentioned in the OP

File: Source/Csla.Blazor/State/NoOpSessionManager.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    { throw new NotSupportedException(); }
+    /// <summary>
+    /// Not supported
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="NotSupportedException"></exception>
+    public Task<Session> RetrieveSession()
+    { throw new NotSupportedException(); }
+    /// <summary>
+    ///
```
[the remark this answers is not in the document]
Comment (swegele) [reply Source/Csla.Blazor/State/NoOpSessionManager.cs:49]: dangit - i'm getting old!

**Review by StefanOssendorf — APPROVED** [review]

**Pull request merged**